animation_nodes/algorithms/mesh/points_scatter.pyx

```diff
@@ -89,19 +101,26 @@ cdef LongList computeDistribution(DoubleList triangleWeights, long numberOfPoint
     outputNumberOfPoints[0] = numberOfPoints
     return distribution
 
-cdef Vector3DList sampleRandomPoints(Vector3DList vertices, PolygonIndicesList polygons, LongList distribution,
-                                     long numberOfPoints, long seed):
+cdef Matrix4x4List sampleRandomPoints(Vector3DList vertices, PolygonIndicesList polygons, Vector3DList polyNormals,
+                                      LongList distribution, long numberOfPoints, long seed):
     cdef XoShiRo256Plus rng = XoShiRo256Plus(seed)
-    cdef Vector3DList points = Vector3DList(length = numberOfPoints)
+    cdef Matrix4x4List matrices = Matrix4x4List(length = numberOfPoints)
+    cdef Vector3 guide = toVector3((0, 0, 1))
```

Comment:
Why initialize the variable here?

Reply:
I thought initializing `toVector3((0, 0, 1))` in loop may affect the performance.

Reply:
Removed.